llama-index-core/llama_index/core/agent/react/step.py

```diff
@@ -689,13 +689,14 @@ async def _arun_step_stream(
 
         # iterate over stream, break out if is final answer after the "Answer: "
         full_response = ChatResponse(
-            message=ChatMessage(content=None, role="assistant")
+            message=ChatMessage(content=None, role=MessageRole.ASSISTANT)
         )
         is_done = False
         async for latest_chunk in chat_stream:
             full_response = latest_chunk
             is_done = self._infer_stream_chunk_is_final(latest_chunk)
             if is_done:
+                full_response.delta = full_response.message.content
```

Comment:
I'm kind of confused what's happening here

The delta should only have the newest tokens from the stream, not the entire response?

I haven't noticed issues with the react agent dropping tokens, can you give some example that shows the issue?

Reply:
tl;dr there was a bit of nuance I missed in debugging the issue. Responded here with more detail/context 
https://github.com/run-llama/llama_index/issues/14307#issuecomment-2186827817